pyogrio/_io.pyx

```diff
@@ -1906,22 +1940,27 @@ def ogr_write(
         gdal_tz_offsets = {}
 
     ### Setup up dataset and layer
-    if not encoding:
-        encoding = locale.getpreferredencoding()
-
     layer_created = create_ogr_dataset_layer(
```

Comment:
This should be a change (or fix) in behaviour for Shapefiles, given that now with this PR, if encoding is not set, it will always use 'ISO-8859-1' (as fixed in `get_encoding_for_driver`)? While before it would have used UTF8 (or cp1252 on Windows) from `locale.getpreferredencoding()`?

(not saying is wrong or so, just trying to understand the code and the change ;))

Reply:
Yes indeed. Not sure if this is explicitly wanted, but this is [what fiona does](https://github.com/Toblerity/Fiona/blob/21ec544adceb95622de026eb06944e6671fd4d76/fiona/ogrext.pyx#L668) (if I understand correctly) and is also what ogr2ogr does based on the [shapefile driver docs](https://gdal.org/drivers/vector/shapefile.html#layer-creation-options).